src/test/java/uk/gov/companieshouse/api/testdata/service/impl/TestDataServiceImplTest.java

```diff
@@ -1001,15 +1005,108 @@ void createAcspMembersDataWithDefaultProfileValues() throws DataException {
+
+        AcspMembersData result = testDataService.createAcspMembersData(spec);
+
+        assertNotNull(result);
+        assertEquals("acspNumber", result.getAcspNumber());
+        assertEquals(acspStatus, spec.getAcspProfile().getStatus());
+        assertEquals(acspType, spec.getAcspProfile().getType());
+        assertEquals(supervisoryBody,
+                spec.getAcspProfile().getAmlDetails().getFirst().getSupervisoryBody());
+        assertEquals(membershipDetails,
+                spec.getAcspProfile().getAmlDetails().getFirst().getMembershipDetails());
+        assertEquals("memberId", result.getAcspMemberId());
+        assertEquals("userId", result.getUserId());
+        assertEquals("active", result.getStatus());
+        assertEquals("role", result.getUserRole());
+
+        verify(acspProfileService).create(argThat(profile ->
+                acspType.equals(profile.getType()) && acspStatus.equals(profile.getStatus())
+                        && profile.getAmlDetails() != null && !profile.getAmlDetails().isEmpty()
+                        && supervisoryBody.equals(
+                                profile.getAmlDetails().getFirst().getSupervisoryBody())
+                        && membershipDetails.equals(
+                                profile.getAmlDetails().getFirst().getMembershipDetails())
+        ));
+
+        verify(acspMembersService).create(argThat(membersSpec ->
+                acspMembersData.getUserId().equals(membersSpec.getUserId())
+        ));
+    }
+
+    @Test
+    void createAcspMembersDataWhenProfileIsNull() throws DataException {
+        AcspMembersSpec spec = new AcspMembersSpec();
+        spec.setUserId("userId");
+
+        AcspProfileData acspProfileData = new AcspProfileData("acspNumber");
+        AcspMembersData acspMembersData;
+        acspMembersData = new AcspMembersData("memberId", "acspNumber", "userId", "active", "role");
+
+        spec.setAcspProfile(null);
+
+        when(acspProfileService.create(any(AcspProfileSpec.class))).thenReturn(acspProfileData);
+        when(acspMembersService.create(any(AcspMembersSpec.class))).thenReturn(acspMembersData);
+
+        AcspMembersData result = testDataService.createAcspMembersData(spec);
+
+        assertNotNull(result);
+        assertEquals("acspNumber", result.getAcspNumber());
+        assertEquals("memberId", result.getAcspMemberId());
+        assertEquals("userId", result.getUserId());
+        assertEquals("active", result.getStatus());
+        assertEquals("role", result.getUserRole());
+
+        verify(acspProfileService).create(argThat(profile ->
+                profile.getStatus() == null
+                        && profile.getType() == null
+                        && profile.getAmlDetails() == null));
+
+        verify(acspMembersService).create(argThat(membersSpec ->
+                acspMembersData.getUserId().equals(membersSpec.getUserId())
```

Comment:
Should we also check the acsp number is been set to the right value in the spec?